src/main/java/org/spongepowered/api/entity/ai/task/builtin/VanillaAITask.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.entity.ai.task.builtin;
+
+import org.spongepowered.api.entity.ai.task.AITask;
+import org.spongepowered.api.entity.living.Agent;
+
+/**
+ * Represents an {@link AITask} that exists in vanilla minecraft.
+ *
+ * @param <O> The type of agent
+ */
+public interface VanillaAITask<O extends Agent> extends AITask<O> {
```

Comment:
Add an `@Deprecated`since this is basically raw NMS behind the scenes.
Also add a setter here otherwise you cannot access/set the underlying mutex bits for custom impls.
Avoid extending this interface in the API even for the built in AITasks. That way we reduce/avoid breaking changes if we find a better solution to this issue.

Maybe you can also add a static helper method 
`static haveMutexClash(AITask<O>, AITask<O> )` (if either of them does not implement VanillaAITask return false)
and 
`static haveMutexClash(VanillaAITask<O>, VanillaAITask<O> )`

so we can do
````java
boolean canRunConcurrentWith(AITask<O> other) {
    return VanillaAITask.haveMutexClash(this. other);
}
````
and also add the constants here again so you can use them with the setter.